src/dune_lang/string_with_vars.mli

```diff
@@ -39,6 +39,16 @@ val virt_pform : ?quoted:bool -> string * int * int * int -> Pform.t -> t
 val virt_text : string * int * int * int -> string -> t
 val make_pform : ?quoted:bool -> Loc.t -> Pform.t -> t
 val make_text : ?quoted:bool -> Loc.t -> string -> t
+
+(** Concatenate a list of parts separate by [sep] which is the empty string by
+    default. *)
+val make
+  :  ?sep:string
+  -> ?quoted:bool
+  -> Loc.t
+  -> [ `Text of string | `Pform of Pform.t ] list
```

Comment:
I find this API a bit overloaded, as it mixes together concatenation of strings with implicit separators and creation of `String_with_vars.t` from a list of different things. I would have expected this to be multiple functions, like `make_text`, `make_pform` and a concatenation function (with possibly an explicit separator) to create a concatenated `String_with_vars.t`.

Reply:
Concatenating a pair of `String_with_vars.t`s is tricky because in addition to being a list of parts (strings and pforms) they also consist of a `Loc.t` (which could just be combined with `Loc.span`) but also a whether the string is quoted which is less clear how to combine. The full type is:
```ocaml
type t =
  { quoted : bool
  ; parts : part list
  ; loc : Loc.t
  }
```
If we were to make a generic concatenation function for `String_with_vars.t` we'd need to come up with a policy for how to merge the `quoted` field so it seemed simpler to let users construct fresh values out of parts.

The `sep` argument is left over from a previous use case of this function that I ended up not using so I'll get rid of it.